MSAL/src/native_auth/public/state_machine/state/MFAStates.swift

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+import Foundation
+
+///  An object of this type is created whenever a user needs to make a specific request to send the MFA challenge.
+@objcMembers
+public class AwaitingMFAState: MSALNativeAuthBaseState {
+
+    /// Requests the server to send the challenge to the default authentication method.
+    /// - Parameter delegate: Delegate that receives callbacks for the operation.
+    public func sendChallenge(delegate: MFASendChallengeDelegate) {
+        // TODO: remove this dummy logic once business logic will be available
+        DispatchQueue.main.async {
+            delegate.onMFASendChallengeVerificationRequired?(newState: 
+                                                                MFARequiredState(
+                                                                    continuationToken: "CT",
+                                                                    correlationId: UUID()
+                                                                ),
+                                                             sentTo: "co****@***.com",
+                                                             channelTargetType: MSALNativeAuthChannelType(value: "email"),
+                                                             codeLength: 8
+            )
+        }
+    }
+}
+
+@objcMembers
+public class MFARequiredState: MSALNativeAuthBaseState {
+
+    /// Requests the server to send the challenge to the specified auth method or the default one.
+    /// - Parameters:
+    ///   - authMethod: Optional. The authentication method you want to use for sending the challenge
+    ///   - delegate: Delegate that receives callbacks for the operation.
```

Comment:
I guess this should have already been discussed during the doc design, but since I was off, is there a special reason for this public method to be called `sendChallenge()`? I think `sendChallenge` and `submitChallenge` might confuse some developers, since "send" can work in both ways (to trigger the backend to send the challenge, or to send the code to the backend)

What about using `requestChallenge()`, or `triggerChallenge()` instead?